dockerfile_scripts/install_libnss_determined.sh

```diff
@@ -5,5 +5,8 @@ set -e
 # Add a plugin to the user system that lets us extend the users available in
 # the container at runtime. This is critical for supporting non-root shell,
 # which in turn is critical for non-root distributed training.
+# Also, place `determined` NSS plug-in to be searched first instead of last so
+# Determined-defined entries are used when there are conflicts with infrastructure-
+# based entries in these identity databases: passwd, shadow, group
 make -C /tmp/det_dockerfile_scripts/libnss_determined libnss_determined.so.2 install \
-    && sed -E -i -e 's/^((passwd|shadow|group):.*)/\1 determined/' /etc/nsswitch.conf
+    && sed -i -E '/^(passwd|group|shadow):/s/[ \t]determined//;/^(passwd|group|shadow):/s/:([ \t]+)/:\1determined /' /etc/nsswitch.conf
```

Comment:
What's wrong with the simpler script I suggested?

```bash
sed -i -E 's/^(group|passwd|shadow): /\1: determined /'
```

Yours is complicated enough I'd have a lot of trouble understanding what I was reading, but I'm not sure I why it is better or safer?

In fact... now that I look closer at it, I think the `s/[ \t]determined//;` bit is unnecessary, since that is to delete a `determined` entry, but there shouldn't be one in a vanilla image.  That was something we did when we were modifying Determined images.

Reply:
Given that it's in single quotes, the one-liner could be split up on the semicolons to improve readability.  And really, the two substitutions should probably be grouped into a block to avoid duplication since they're using the same line selector.

```suggestion
    && sed -i -E '
    /^(passwd|group|shadow):/{
      s/[ \t]determined//;
      s/:([ \t]+)/:\1determined /;
    }' /etc/nsswitch.conf
```

Reply:
(actually, I'm no longer positive about the `\s` - that may or may not be universal.  I'd probably actually use `[[:space:]]`, but :shrug:

Reply:
Amending after @dannysauer's use of `{cmd;cmd}` syntax which, until today, I was unware 🤦🏻‍♂️ 

`/^(passwd|group|shadow):/{s/[ \t]determined//;s/:([ \t]+)/:\1determined /}`

1. `/^(passwd|group|shadow):/` Run substitute only the lines beginning with passwd, group, shadow; none of the others
2. `s/[ \t]determined//;` Remove "determined", and leading space|tab, if it happens to already exist
3. `s/:([ \t]+)/:\1determined /` Insert "determined" after all intervening spaces and tabs between the colon after the database name and the first _source_

@rb-determined-ai, the reason for this particular sed script is I get nervous if start accepting too many assumptions about what I'm updating.  If I think of something that can break my algorithm, I try to prevent it provided the solution is reasonable and feasible ... and maintainable.  Danny proposed a readability enhancement without reducing its functionality.